kafka-junit-core/src/main/java/com/salesforce/kafka/test/KafkaTestServer.java

```diff
@@ -76,19 +77,44 @@
      */
     private final String localHostname;
 
+    /**
+     * Defines overridden broker properties.
+     */
+    private final Properties overrideBrokerProperties = new Properties();
+
     /**
      * Default constructor using "127.0.0.1" as the advertised host.
      */
     public KafkaTestServer() {
-        this("127.0.0.1");
+        this("127.0.0.1", new Properties());
     }
 
     /**
      * Alternative constructor allowing override of advertised host.
      * @param localHostname What IP or hostname to advertise services on.
      */
     public KafkaTestServer(final String localHostname) {
+        this(localHostname, new Properties());
+    }
+
+    /**
+     * Alternative constructor allowing override of advertised host and brokerProperties.
+     * @param overrideBrokerProperties Define Kafka broker properties.
+     */
+    public KafkaTestServer(final Properties overrideBrokerProperties) {
+        this("127.0.0.1", overrideBrokerProperties);
+    }
+
+    /**
+     * Alternative constructor allowing override of advertised host and brokerProperties.
+     * @param localHostname What IP or hostname to advertise services on.
+     * @param overrideBrokerProperties Define Kafka broker properties.
+     */
+    public KafkaTestServer(final String localHostname, final Properties overrideBrokerProperties) {
```

Comment:
I kind of feel like you should pull hostname from the supplied properties rather than take this in as a parameter.

Reply:
Done!  To avoid breaking backwards compatibility I've left this constructor, but marked it deprecated.